core/sys/linux/sys.odin

```diff
@@ -729,18 +729,32 @@ getsockopt :: proc {
 	getsockopt_base,
 }
 
-// TODO(flysand): clone (probably not in this PR, maybe not ever)
+/*
+	Creates a new ("child") process, in a manner similar to fork.
+	Available since Linux 1.0? (<2.4)
+
+	Note(flysand): this syscall is not documented, but the bottom 8 bits of flags
+	are for exit signal
+*/
+clone :: proc "contextless" (flags: u64, stack: rawptr, parent_tid, child_tid: ^i32, tls: u64) -> (i64, Errno) {
+	when ODIN_ARCH == .amd64 {
+		ret := syscall(SYS_clone, flags, stack, parent_tid, child_tid, tls)
+		return errno_unwrap(ret, i64)
```

Comment:
I'll explain why I didn't want to make clone syscall part of the API. If one calls this function, providing a `stack` parameter that is not NULL, then this line will crash the child process, because child's stack doesn't have the value for RIP on the stack.

Usually this syscall is used for creating threads, and for threads the ideal API is a callback. The way one would set this up is by calling the `clone` syscall directly, and if we found ourselves as the child process, we jump to (not `call`) an assembly routine that sets up the stack, copies the TLS image and does a few things, before calling the thread callback with needed parameters.

But calling a callback, let alone having a separate assembly routine is too much of an API change, which goes against the way this package is supposed to work. That's why I didn't want to provide clone as part of this sys/linux API.

Even if you manage to create a thread you need to know the exact TLS layout for the given libc the program is compiled with. And the TLS is defined by the dynamic linker, which makes things so much harder because there's no API's that let you get the TLS image / struct from a dynamic linker and call it done. Having a different TLS layout can lead to bugs, from what I've heard nvidia drivers on linux expect glibc's TLS layout and don't work with e.g. musl.

So I think this procedure needs to be removed and syscall called directly in the `fork()` procedure.